device/core/src/blob_upload/blob_upload_result.ts

```diff
@@ -40,30 +40,41 @@ export class BlobUploadResult {
   /*Codes_SRS_NODE_DEVICE_BLOB_UPLOAD_RESULT_16_007: [If the `err` argument is not `null` but body and response are `undefined` (non HTTP error), the `BlobUploadResult` error shall have the `statusCode` property set to -1.]*/
   /*Codes_SRS_NODE_DEVICE_BLOB_UPLOAD_RESULT_16_008: [If the `err` argument is not `null` but body and response are `undefined` (non HTTP error), the `BlobUploadResult` error shall have the `statusDescription` property set to the error message.]*/
-  /*Codes_SRS_NODE_DEVICE_BLOB_UPLOAD_RESULT_16_009: [If `err` is null but `body` and `reponse` are provided, the `BlobUploadResult` error shall have the `isSuccess` property set to `false`.]*/
+  /*Codes_SRS_NODE_DEVICE_BLOB_UPLOAD_RESULT_16_009: [If `err` is null but `body` and `reponse` are provided, the `BlobUploadResult` error shall have the `isSuccess` property set to `true`.]*/
   /*Codes_SRS_NODE_DEVICE_BLOB_UPLOAD_RESULT_16_010: [If `err` is null but `body` and `reponse` are provided, the `BlobUploadResult` error shall have the `statusCode` property set to the HTTP status code of the blob upload response.]*/
   /*Codes_SRS_NODE_DEVICE_BLOB_UPLOAD_RESULT_16_011: [If `err` is null but `body` and `reponse` are provided, the `BlobUploadResult` error shall have the `statusDescription` property set to the HTTP response body of the blob upload response.]*/
   /*Codes_SRS_NODE_DEVICE_BLOB_UPLOAD_RESULT_16_012: [If `err` is null and `body` and/or `response` is/are falsy, `fromAzureStorageCallbackArgs` shall throw a `ReferenceError`.]*/
-  static fromAzureStorageCallbackArgs(err?: Error, body?: any, response?: BlobResponse): BlobUploadResult {
-    if (!err && (!body || !response)) throw new ReferenceError('if err is null, body and response must be supplied');
+  static fromAzureStorageCallbackArgs(err?: Error & RestErrorStub, uploadResponse?: any): BlobUploadResult {
     let uploadResult: BlobUploadResult;
+    if ((err && uploadResponse) || (!err && !uploadResponse)) throw new ReferenceError('either err or uploadResponse must be supplied, exclusively.');
+    // from new storage API
     if (err) {
-      const statusCode = response ? response.statusCode : -1;
-      const statusDescription = response ? response.body : err.message;
+      const statusCode = err.hasOwnProperty('statusCode') ? err.statusCode : -1;
+      const statusDescription = err.hasOwnProperty('response') ? err.response : err.message;
       uploadResult = new BlobUploadResult(false, statusCode, statusDescription);
     } else {
-        if (response.statusCode >= 200 && response.statusCode < 300) {
-          uploadResult = new BlobUploadResult(true, response.statusCode, response.body);
-        } else {
-          uploadResult = new BlobUploadResult(false, response.statusCode, response.body);
-        }
+      if (!uploadResponse._response) throw new ReferenceError('if uploadResponse is provided, _response must be a member object');
```

Comment:
>throw [](start = 37, length = 5)

just want to make sure: we throw only when it's a developer error (ie a bug in our SDK or in the customer code). If it's something that could happen because a call to the service fails, either it shouldn't throw, or it should throw but we catch it and call the public api callback with the error.